### PR TITLE
Add julia examples

### PR DESCRIPTION
**Pull request by dannys4**
These are intended to be paired with the _second_ PR for Julia bindings to make things less clunky

### CODE REVIEW

File: examples/julia/FromDensity1D.jl [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+##
+A = reshape(0:1,2,1) # The ;; makes a single row matrix then transpose for single column
+mset = MultiIndexSet(A)
+fixed_mset = Fix(mset, true)
+opts = MapOptions()
+monotoneMap = CreateComponent(fixed_mset, opts)
+
+# KL divergence objective
+function objective(coeffs,_)
```
Comment (bonh) [inline examples/julia/FromDensity1D.jl:38]: Perhaps it is different for Julia but I prefer functions that are self-contained without variables implicitly fetched from a top-level scope.

File: examples/julia/FromSamples2D_banana.jl [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+x = collect([x1 x2]')
+
+test_n_pts = 10_000
+test_z = randn(2,test_n_pts)
+test_x1 = test_z[1,:]
+test_x2 = test_z[2,:] + test_z[1,:].^2
+test_x = collect([test_x1 test_x2]')
+
+# For Plotting and computing reference density
+rho = MvNormal(I(2))
```
Comment (bonh) [inline examples/julia/FromSamples2D_banana.jl:23]: Michael started to rename `rho` to the more explicit `reference_density` or something like that...

**Review by bonh — COMMENTED** [review]
Looks good, few minor issues. Thanks!

**Comment by bonh** [comment]
Oh, where is that second PR?

**Comment by dannys4** [comment]
What second PR?

File: examples/julia/FromDensity1D.jl (hunk 1 above)
Comment (dannys4) [reply examples/julia/FromDensity1D.jl:38]: I might agree, but the python bindings have a weird amalgamation of some data being fetched from top level scope and some being local scope, which confuses the heck out of me, so I just did it consistently. I can change this pretty easily though.

File: examples/julia/FromSamples2D_banana.jl (hunk 2 above)
Comment (dannys4) [reply examples/julia/FromSamples2D_banana.jl:23]: Got it, can do

File: examples/julia/FromDensity1D.jl (hunk 1 above)
Comment (bonh) [reply examples/julia/FromDensity1D.jl:38]: Do you mean the python examples? We tried to use only local data. Can you point out which examples still use a top level scope in their functions?

**Comment by bonh** [comment]
At the top you write 
> These are intended to be paired with the second PR for Julia bindings to make things less clunky.

**Comment by dannys4** [comment]
Ah, yes, sorry, the second PR in the main repository for julia bindings which had not been merged at the time of this PR in the examples repo. Now that it's been merged, you can ignore that comment.

File: examples/julia/FromDensity1D.jl (hunk 1 above)
Comment (dannys4) [reply examples/julia/FromDensity1D.jl:38]: Sorry, it looks like I was confused or was looking at an old copy when I saw that. I'll get around to that shortly.

Comment (bonh) [reply examples/julia/FromDensity1D.jl:38]: Or we can merge that for now and fix the cosmetic issues later, after we have finished the examples In Python, right?

Comment (dannys4) [reply examples/julia/FromDensity1D.jl:38]: Okay actually, looking back through, I did note that FromSamples [uses `rv`](https://github.com/MeasureTransport/MParT-examples/blob/4b196a82d593438ff9c8c85ae37abc8b20016c52/examples/python/FromSamples1D.py#L57) as a local variable sometimes, if you'd like to put up a small PR on that at some point.

**Comment by dannys4** [comment]
@bonh Sorry for the delay, but I pushed up changes that should fix what you're looking for

File: examples/julia/FromDensity1D.jl (hunk 1 above)
Comment (bonh) [reply examples/julia/FromDensity1D.jl]: Thanks! I have added it to #15

**Review by bonh — APPROVED** [review]

**Pull request merged**